Merge to main

EPKgit opened the pull request.

EPKgit committed:
Fix collision equations

EPKgit committed:
Time slow on death

EPKgit committed:
Stats changes

EPKgit committed:
Player stats tuning, Bombs now ramp up before becoming active

EPKgit committed:
network testing

EPKgit committed:
Photon Testing

EPKgit committed:
Basic player movement prediction

EPKgit committed:
Revert "Player stats tuning, Bombs now ramp up before becoming active"

This reverts commit 5cb542f4fe9512e3165c0706fd7154988b6c0402.

EPKgit committed:
Revert "Revert "Player stats tuning, Bombs now ramp up before becoming active""

This reverts commit ab11aa0e2ce21223aea31f0a974f89a826f6ac20.

EPKgit committed:
Revert "Basic player movement prediction"

This reverts commit 3c3958eaec37d20b984b3bc3dc9793fe11e8118d.

EPKgit committed:
Revert "Photon Testing"

This reverts commit 4d2f9421d8e5043154d86926cffb5254efe202bc.

EPKgit committed:
Revert "network testing"

This reverts commit 71562129a8fae27d44a6dd0a7db71e1287cf9c42.

syy1125 commented on Assets/Scripts/CurveTable.cs:
```diff
@@ -0,0 +1,10 @@
+{
+	public static float[] Power = { 1.0f, 1.1f, 1.2f };
+	public static float[] Resistance = { 1.0f, 1.1f, 1.2f };
+	public static float[] Acceleration = { 1.33f, 1.66f, 2.0f };
```
My concern is that these changes would be too strong a nerf - having power 3 = 1.2x the opponent's velocity feels a little weak.
That said, we can implement these changes for now and see how the balance works out, then buff and nerf as necessary.

EPKgit replied:
I've been testing with these values as of current and they seem to be fine, but they can be changed, I just wanted to consolidate them in one place to make that an easy change.

syy1125 commented on Assets/Scripts/PlayerCollision.cs:
```diff
@@ -26,53 +50,53 @@ private void OnCollisionEnter2D(Collision2D col)
+			Feedback(other, _rigidbody2D.velocity.magnitude + rb.velocity.magnitude);
+			float a1 = Vector2.Angle(rb.velocity, transform.position - other.transform.position);
+			float a2 = Vector2.Angle(_rigidbody2D.velocity, transform.position - other.transform.position);
+			if(a1 > a2)
```
So after reading what `DoCollision` does it looks like there isn't any difference between who is `instigator` and who is `receiver`. Am I correct in that respect?
If I am correct, what is the purpose of doing this unnecessary computation?

EPKgit replied:
This is not currently being used, but will be used in the future for networking collisions as only one of the players should do the collision calculation. It also opens design space for abilities relying one being one or the other.

syy1125 commented on Assets/Scripts/PlayerCollision.cs:
```diff
@@ -26,53 +50,53 @@ private void OnCollisionEnter2D(Collision2D col)
 		AudioManager.instance.PlayClangSound();
-		_particleSystem.transform.position = toSpawn;
 		Camera.main.GetComponent<CameraController>().ScreenShake(totalMagnitude);
+		_particleSystem.transform.position = (other.transform.position + gameObject.transform.position) / 2;
```
This doesn't fix the issue with collisions teleporting player 1 to the point of contact.
In combination with the `OnTriggerStay2D` above, this could result in uncontrollable collisions.

EPKgit replied:
The ontriggerstay2d was an attempt to fix when two players spawn on top of eachother or otherwise lerp into one another somehow. The particle system teleportation can be removed seperately if it's still an issue

syy1125 replied:
I'm going to try playing with this and see what happens.

syy1125 replied:
I just tried out the collision. It's weird. By no means was it uncontrollable, but definitely feels wrong. I'll post a gif to discord in a moment.

EPKgit committed:
Fix players teleporting into one another on collison, moved spark effect from player to a seperate prefab

EPKgit committed:
Update Assets/Prefabs/PlayerSparkPrefab.prefab

syy1125 committed:
Tweak collision strength

Collision feels way too strong.
Player relative direction is now normalized.
Increase collision multiplier to keep collisions strong. (was 0.9, now 1.5)

syy1125 reviewed (approved):
Ok LGTM
